### PR TITLE
Multi-Class Image Classifier using Deep AutoViML

### LINKED ISSUE #6231: [Machine Learning] Multi-Class Image Classifier using Deep AutoViML
#### Proposed title of article
[Machine Learning] Multi-Class Image Classifier using Deep AutoViML

#### Proposed article introduction
Image classification is one of the supervised machine learning problems which aims to categorize the images of a dataset into their respective categories or labels. Multi-Class image classification is a task in computer vision, where we categorize an image into three or more classes.  Deep Autoviml is an AutoML library for building deep learning models using TensorFlow and Keras.

Deep AutoViML will automatically load a wide variety of performant DNN architectures such as deep and wide, deep and cross models. The loaded model will then be used for image classification. Deep AutoViML will also automatically select the best model, add pre-processing layers for feature transformation and do selective feature engineering.

In this tutorial, we will use Deep AutoViML to build a custom model that can identify images of hands playing rock-paper-scissors games. Our image will be made up of three classes.

#### Key takeaways 
1. What is multi-class image classification.
2. Installing Deep AutoViML.
3. Image dataset pre-processing.
4. Using Deep AutoViML to automatically select the best model.
5. Fine-tuning the model to give the best prediction results.
#### Article quality
This tutorial is unique because we will compare multi-label and multi-class image classification. This will give the reader a deeper understanding of multi-class image classification and how to differentiate the two classification types. The tutorial will also explain in details the techniques that AutoViMl uses to select the best model for image selection. We will explore how  AutoViML searches and fine-tunes the neural network layers to get the best model architecture. By the end, we will have a fully optimized neural network that will give the best results.

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by simonndiritu477**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6231

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @simonndiritu477 ,

Welcome to EngEd program!

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you to get started with the review.

Meanwhile, we suggest you to go through approved PRs/published articles to see if you have missed out anything. This would help us avoid any potential closure of the PR.

Thanks.

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Awesome article @simonndiritu477, just a few suggestions as we move on to tech review. I noticed some of your images are quite large. We should try to make sure our images are of the size 600x400 pixels. 

Kindly resize the necessary images using the [Pixlr Image Editor](https://pixlr.com/e). Also, make sure to list sources to any images that have been sourced from other sites. Ping me when done.

**Comment by simonndiritu477** [comment]
Hello @Dawe-7, I have resized some of the images. But, these images are screenshots of the model's outputs (apart from the hero image), if I further resize most of them to 600X400 px they will not show the desired results.
Thanks.

**Review by Dawe-7 — APPROVED** [review]
Hello @simonndiritu477 , welcome to Section's EngEd and congratulations on submitting your first article. The article is an interesting read and easy to follow through. Nice job!

Thank you for submitting an article that is well polished, this article is ready for final review. Kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles.  A final review will be done by a Content Moderator. 

In case of any question or clarification (currently or in the future), do not hesitate to reach out via Slack (Marsa) or my personal email [dantedawek@gmail.com](mailto:dantedawek@gmail.com).
I am ALWAYS here to help.

Looking forward to seeing you contribute more to the EngEd program.

**Comment by hectorkambow** [comment]
jumping on this one

**Comment by hectorkambow** [comment]
@simonndiritu477 

image


cc @Dawe-7 

FYI not sure why I could not push my commits to this PR.
